lib/service/service.go

```diff
@@ -1217,6 +1217,10 @@ func (process *TeleportProcess) initAuthService() error {
 		log.Errorf("PID: %v Failed to bind to address %v: %v, exiting.", os.Getpid(), cfg.Auth.SSHAddr.Addr, err)
 		return trace.Wrap(err)
 	}
+
+	authSSHAddr := cfg.Auth.SSHAddr
+	authSSHAddr.Addr = listener.Addr().String()
```

Comment:
Use the string addr directly, since that's all you need.
`utils.NetAddr` is basically a thin wrapper around a string anyway.
```suggestion
	authSSHAddr := listener.Addr().String()
```